internal/handler/ticket.go

```diff
@@ -0,0 +1,51 @@
+package handler
+
+import (
+	"context"
+	"log/slog"
+	"net/http"
+
+	"github.com/dilyara4949/flight-booking-api/internal/domain"
+	"github.com/dilyara4949/flight-booking-api/internal/handler/request"
+	"github.com/dilyara4949/flight-booking-api/internal/handler/response"
+	"github.com/dilyara4949/flight-booking-api/internal/middleware"
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+)
+
+type TicketService interface {
+	Update(ctx context.Context, ticketID, userID uuid.UUID, req request.UpdateTicket) (domain.Ticket, error)
+}
+
+func UpdateTicketHandler(service TicketService) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var req request.UpdateTicket
```

Comment:
don't forget access check

Reply:
didn't check it since i am taking user id from context, and it will be updated only if ticket's userId will match id from context:
```go
if err := repo.db.WithContext(ctx).Model(&ticket).
		Where("id = ? AND user_id = ?", ticket.ID, ticket.UserID).
		Updates(map[string]interface{}{
			"price":      ticket.Price,
			"updated_at": time.Now(),
		}).Error; err != nil {
		return ticket, fmt.Errorf("update ticket error: %w", err)}
```

Reply:
In that case, how somebody with Admin access will be able to update another user's ticket?